### DISCUSSION TITLE
Will a constraint not consume CUP time when it with a zero constraint weight?

### DISCUSSION THREAD

**Discussion by kentbill** [TimefoldAI/timefold-solver#472] [General]
Hi, teams,
From the following code, does it mean a constraint wich with a zero constraint weight will be ignored? These constraints will not be included in the score calculation? 

[https://github.com/TimefoldAI/timefold-solver/blob/8768155a076de1f4b8537fa2ebb62811562e9c67/core/constraint-streams/src/main/java/ai/timefold/solver/constraint/streams/bavet/BavetConstraintSessionFactory.java#L60](url)

**Comment by triceo** [comment]
Exactly. When a constraint's weight is set to zero, it is completely eliminated from processing, as if it never existed in the first place.

**Reply by kentbill** [reply]
Great, thank you.
We can make constraint customization (such as enable/disable a constraint) more concise through the constraint weight.
